internal/provider/provider.go

```diff
@@ -64,8 +68,87 @@ func New() func() *schema.Provider {
 	}
 }
 
+const statuspageUrl = "https://pdrzb3d64wsj.statuspage.io/api/v2/components.json"
+const statuspageHcpComponentId = "ym75hzpmfq4q"
+
+type status string
+
+// Possible statuses returned by statuspage.io.
+const (
+	operational         status = "operational"
+	degradedPerformance        = "degraded_performance"
+	partialOutage              = "partial_outage"
+	majorOutage                = "major_outage"
+	underMaintenance           = "under_maintenance"
+)
+
+type statuspage struct {
+	Components []component `json:"components"`
+}
+
+type component struct {
+	ID     string `json:"id"`
+	Status status `json:"status"`
+}
+
 func configure(p *schema.Provider) func(context.Context, *schema.ResourceData) (interface{}, diag.Diagnostics) {
 	return func(ctx context.Context, d *schema.ResourceData) (interface{}, diag.Diagnostics) {
+
+		// Check the Hashicorp status page in case there's an outage.
```

Comment:
Nit: Maybe move this functionality out to a separate file, and then just call something like `bool, diags = statuspage.isHCPOperational()`. Mainly suggesting this so that the `provider/provider.go` doesn't become unwieldy as more functionality is added.

Reply:
Yeah I like that plan